libbindgen/src/codegen/mod.rs

```diff
@@ -516,6 +516,30 @@ impl CodeGenerator for Type {
                     inner_item.to_rust_ty(ctx)
                 };
 
+                {
+                    // FIXME(emilio): This is a workaround to avoid generating
+                    // incorrect type aliases because of types that we haven't
+                    // been able to resolve (because, eg, they depend on a
+                    // template parameter).
+                    //
+                    // It's kind of a shame not generating them even when they
+                    // could be referenced, but we already do the same for items
+                    // with invalid template parameters, and at least this way
+                    // they can be replaced, instead of generating plain invalid
+                    // code.
+                    let inner_canon_type =
+                        inner_item.expect_type().canonical_type(ctx);
+                    if inner_canon_type.is_named() &&
+                        inner_canon_type.name().map_or(false, |name| {
+                            name.contains("decltype ") ||
+                            name.contains("decltype(")
+                        }) {
```

Comment:
I tend to agree with this. My thinking was that given this is the exception, and not the norm, I think we want to know when can we generate these incorrect bits before skipping them.

I'm fine tweaking this to be a generic identifier check though, I guess.